lte/gateway/c/session_manager/AmfServiceClient.cpp

```diff
@@ -0,0 +1,52 @@
+#include "AmfServiceClient.h"
+#include "ServiceRegistrySingleton.h"
+#include "magma_logging.h"
+
+// using google::protobuf::RepeatedPtrField;
+using grpc::Status;
+
+namespace {
+
+void call_back_void_amf(grpc::Status, magma::SmContextVoid respvoid)
+{
+  //do nothinf but to only passing call back
+  //cout <<" Only for testing call back" << endl;
+}
+std::function<void(grpc::Status, magma::SmContextVoid)>callback  = call_back_void_amf;
+
+}
+
+namespace magma {
+
+AsyncAmfServiceClient::AsyncAmfServiceClient(
+    std::shared_ptr<grpc::Channel> channel)
+    : stub_(SmfPduSessionSmContext::NewStub(channel)) {}
+
+AsyncAmfServiceClient::AsyncAmfServiceClient()
+    : AsyncAmfServiceClient(
+          ServiceRegistrySingleton::Instance()->GetGrpcChannel(
+              "spgw_service", ServiceRegistrySingleton::LOCAL)) {}
+
+
+bool AsyncAmfServiceClient::handle_response_to_access(
+    const magma::SetSMSessionContextAccess& response) {
+  
+  MLOG(MINFO) << "Sending Set SM Session Response from SMF ";
+  handle_response_to_access_rpc(response, callback);
```

Comment:
Is this function useful if it only calls another function? Or do you have plans for expanding this function to do more things?

Reply:
will comeback on this.

Reply:
It is needed. In future we need to categorize response of establishment and modification. Off-course signature will be modified then.